src/main/java/com/google/sps/travelbud/CityServlet.java

```diff
@@ -25,42 +31,54 @@
 @WebServlet(urlPatterns = {"/api/cities/*"})
 public class CityServlet extends HttpServlet {
   @Override
-
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json;");
 
-    // list of cities
-    List<City> cities = City.CITIES;
-
     String endpoint = request.getPathInfo();
 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
     Gson gson = new Gson();
 
     if (endpoint == null) {
       // return the list of cities
+      List<City> cities = City.getAll(datastore);
       response.getWriter().println(gson.toJson(cities));
 
     } else {
-      City city = getCity(cities, endpoint);
+      // adjust endpoint
+      long cityId = Long.parseLong(endpoint.substring(1));
+      City city = City.getCity(datastore, cityId);
       if (city == null) {
         response.sendError(HttpServletResponse.SC_NOT_FOUND, "City Not Found");
       } else {
         response.getWriter().println(gson.toJson(city));
       }
     }
   }
-  public City getCity(List<City> cities, String path) {
-    // adjust endpoint
-    int cityId = Integer.parseInt(path.substring(1));
 
-    // search for each country element in the List
-    for (City tempCity : cities) {
-      int tempCityId = tempCity.getId();
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    long id = Long.parseLong(getParameter(request, "id", ""));
+    String name = getParameter(request, "name", "");
+    long countryId = Long.parseLong(getParameter(request, "countryId", ""));
+    String description = getParameter(request, "description", "");
 
-      if (tempCityId == cityId) {
-        return tempCity;
-      }
+    Entity cityEntity = new Entity("City", id);
```

Comment:
I suggest not reading an ID from the poster as there's no guarantee it'll be unique. Instead, you can rely on the ID generated when you do the put (see [`Entity(java.lang.String kind)`](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/Entity.html#Entity-java.lang.String-))